web/blog/2022-09-05-dev-excuses-app-tutrial.md

```diff
@@ -0,0 +1,268 @@
+```
+curl -sSL https://get.wasp-lang.dev/installer.sh | sh
+```
+
+Now let’s create a new web app named ItWaspsOnMyMachine.
+```
+wasp new ItWaspsOnMyMachine
+```
+
+Changing the working directory:
+```
+cd ItWaspsOnMyMachine
+```
+
+Starting the app:
+```
+wasp start
+```
+
+Now your default browser should open up with a simple predefined text message. That’s it! 🥳 We’ve built and run a NodeJS + React application. And for now – the codebase consists of only two files! `main.wasp` is the config file that defines the application’s functionality. And `MainPage.js` is the front-end.
+
+![Initial page](../static/img/init-page.png)
+
+
+## That worked perfectly when I developed it.
+
+
+**1) Let’s add some additional configuration to our `main.wasp` file. So it will look like this:**
+
+```js title="main.wasp | Defining Excuse entity, queries and action"
+
+// Main declaration, defines a new web app.
+app ItWaspsOnMyMachine {
+
+  // Used as a browser tab title.                                  
+  title: "It Wasps On My Machine",
+
+  head: [
+    // Adding Tailwind to make our UI prettier
+    "<script src='https://cdn.tailwindcss.com'></script>"
+  ],
+
+  dependencies: [ 
+    // Adding Axios for making HTTP requests                                          
+    ("axios", "^0.21.1")
+  ]
+}
+
+// Render page MainPage on url `/` (default url).
+route RootRoute { path: "/", to: MainPage }                 
+
+// ReactJS implementation of our page located in `ext/MainPage.js` as a default export
+page MainPage {                                             
+  component: import Main from "@ext/MainPage.js"
+}
+
+// Prisma database entity
+entity Excuse {=psl                                          
+    id          Int     @id @default(autoincrement())
+    text        String
+psl=}
+
+// Query declaration to get a new excuse
+query getExcuse {                                           
+  fn: import { getExcuse } from "@ext/queries.js",
+  entities: [Excuse]
+}
+
+// Query declaration to get all excuses
+query getAllSavedExcuses {                                  
+  fn: import { getAllSavedExcuses } from "@ext/queries.js",
+  entities: [Excuse]
+}
+
+// Action to save current excuse
+action saveExcuse {                                         
+  fn: import { saveExcuse } from "@ext/actions.js",
+  entities: [Excuse]
+}
+```
+
+We’ve added Tailwind to make our UI more pretty and Axios for making API requests. 
+
+Also, we’ve declared a database entity called `Excuse`, queries, and action. The `Excuse` entity consists of the entity’s ID and the text. 
+
+`Queries` are here when we need to fetch/read something, while `actions` are here when we need to change/update data. Both query and action declaration consists of two lines – a reference to the file that contains implementation and a data model to operate on. You can find more info in the docs. So let’s proceed with queries/actions. 
```

Comment:
Maybe we should link to those operations docs when we say "You can find more info in the docs"?